### PR TITLE
Enhance/intent classifier

### PR DESCRIPTION
**Pull request by MohamedAliBouhaouala**
# Motivation

This PR enhances the intent classifier for the English language. It also adds some refactoring to the inference function making it usable via command lines instead of loading a json file. 

# Type of change:

Code refactoring to models and data loaders. Enhancing intent classifier by changing the loss function (from a standard cross-entropy loss to focal cross entropy loss: https://paperswithcode.com/method/focal-loss), devising a new metric for calculating the model prediction's certainty.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: nlu/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,37 @@
-tensorflow==2.13.*
-transformers==4.30.2
-keras==2.13.*
-numpy==1.24.*
-scikit_learn==1.2.2
-fastapi==0.100.0
-uvicorn[standard]==0.23.1
-autopep8==2.0.2
-h5py --only-binary=h5py
+absl-py==2.1.0
```
Comment (marrouchi) [inline nlu/requirements.txt:1]: Can we keep the requirement.txt as is if we didn't install any extra packages ?

**Review by marrouchi — CHANGES_REQUESTED** [review]
Looks good 👍  left one comment

File: nlu/requirements.txt (hunk 1 above)
Comment (MohamedAliBouhaouala) [reply nlu/requirements.txt]: Addressed. Only extra package needed is the focal_loss package

**Review by marrouchi — APPROVED** [review]
